iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/methods/DeviceMethodTests.java

```diff
@@ -483,4 +489,39 @@ public void invokeMethodOnUnregisteredDevice() throws Exception
             }
         }
     }
+
+    @Test
+    @StandardTierHubOnlyTest
+    public void invokeMethodWithServiceSideProxy() throws Exception
+    {
+        if (testInstance.protocol != IotHubClientProtocol.MQTT || testInstance.authenticationType != AuthenticationType.SAS)
+        {
+            // This test doesn't really care about the device side protocol or authentication, so just run it once
+            // when the device is using MQTT with SAS auth
+            return;
+        }
+
+        String testProxyHostname = "127.0.0.1";
+        int testProxyPort = 8894;
+        HttpProxyServer proxyServer = DefaultHttpProxyServer.bootstrap()
+                .withPort(testProxyPort)
+                .start();
+
+        try
+        {
+            Proxy serviceSideProxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(testProxyHostname, testProxyPort));
+
+            ProxyOptions proxyOptions = new ProxyOptions(serviceSideProxy);
+            DeviceMethodClientOptions options = DeviceMethodClientOptions.builder().proxyOptions(proxyOptions).build();
+
+            this.testInstance.methodServiceClient = DeviceMethod.createFromConnectionString(iotHubConnectionString, options);
```

Comment:
By default, the test instance already has a methodServiceClient, but it isn't setup to use a proxy, so this will create a proxy enabled client for this test in particular